docs/spec.md

```diff
@@ -6185,6 +6185,16 @@ At the moment, StableHLO does not provide guarantees about numerical accuracy,
 but this may change in the future
 ([#1156](https://github.com/openxla/stablehlo/issues/1156)).
 
+### Execution semantics of quantized operation
+
+The interpretation of quantized StableHLO operations may vary depending on the
+hardware requirements and capabilities. For instance, some hardware may opt to
+interpret quantized operations using a "dequantize, perform floating-point
+operation, and finally quantize" strategy. Others may perform the entire
+computation with integer arithmetic. Consequently, the interpretation of
+quantized StableHLO operations is exclusively determined by the specific
```

Comment:
How does `precision_config` in `dot_general` or `convolution` play with this flexibility? Take a quantized convolution op, for example; if the precision config is `HIGHEST` is it still ok to use integer accumulation?

Related:
* https://github.com/openxla/stablehlo/issues/755
* https://github.com/openxla/stablehlo/blob/main/docs/spec.md#dot_general

Reply:
Hi @dansuh17 
That is a good point.

If the expectation is to have  `HIGHEST` precision, then the implementation can use `f32` or `f64` (as applicable) for computation (using the dequant-op-quant strategy). With that said, for quantized operations, an implementation can choose to a different implementation compared to the naive definition of  precision config (which IMO a trade-off between precison and speed) as long as the semantics of the operations is preserved. We will be able to evaluate the later part (preversing the semantics) with strong numerical accuracy guarentees defined at the StableHLO level (https://github.com/openxla/stablehlo/issues/1156, yet to be defined)

Reply:
IMHO, the concept of quantization contradicts with `precision_config`.  Representing `f32` or `f64` in 8-bit value already throws away a lot of precision bits.  I don't think there are sensible way to combine both of them and provide some kind of guarantees or set some kind of expectations.

Reply:
Thanks @loganchien. I can see your point. 

IMO, we should clarify the coexistence of  quantization operations with precision config. At this point there exist some open questions like (1) if quantization and `precision_config`  should co-exist, and (2) if yes, what is the desired behavior.
 
@dansuh17 
Added an AI @https://github.com/openxla/stablehlo/issues/755#issuecomment-1977199585. How about we take care of this as part of resolving the linked issue?

Reply:
SG, thanks for addressing this. Could you also add in `spec.md` about the current status regarding `precision_config`?

Reply:
Thanks for the approval, @dansuh17 !
 
In the spec we left a comment from the underspecified nature of `precision_config` and referred to #755 at location https://github.com/openxla/stablehlo/blob/main/docs/spec.md#semantics-30.

PTAL.

Reply:
For a quantized implementation of dot_general, where the accumulation is done as integer accumulation followed by a post-scale, I think it can be useful to select the width of the accumulator. For example, for quantized 16-bit by 8-bit dot product operations, both 32-bit and 48-bit accumulation range may be useful. So, I think there is a role for something like precision_config in quantized arithmetic to define the accumulator range.

Reply:
Hi Dominic,
That is a valid point and IMO should be captured as part of speccing the `precision_config`, which, in its current form,  may impact quantized operations in an unspecified way.
How about we take this as part of #755?

Reply:
Hi @sdasgup3 . Yes, it seems fine to take this separately. If I understand correctly this patch is making clear that different types of implementations are possible for quantized in terms of mixed floating-point or full integer. This patch does not define further details on what level of accuracy or range is expected and this would be covered in other issues, for example on precision_config.